libr/util/str.c

```diff
@@ -2350,22 +2350,47 @@ R_API bool r_str_glob(const char* str, const char *glob) {
 	}
 	while (*str) {
 		if (!*glob) {
-			return true;
+			return false;
 		}
 		switch (*glob) {
 		case '*':
 			if (!*++glob) {
 				return true;
 			}
+			// Advance glob an additional time if it is a '**'
+			if (*glob == '*') {
+				if (!*++glob) {
+					return true;
+				}
+			}
+			// Check if there are additional wildcards
+			// if so, we need to search for the substring in between the wildcards
+			const char *needle_end = glob;
+			while (true) {
+				if (*needle_end == '*' ||
+					*needle_end == '?' ||
+					*needle_end == '$' ||
+					*needle_end == '^' ||
+					*needle_end == '\0') {
+					break;
+				}
+				needle_end++;
```

Comment:
if needle_end == 0, then its better not to go foward with this needle_end++ or you are pointing outside the memory